### PR TITLE
Create Robot Structure File

### PR DESCRIPTION
**Pull request by Twinters007**

### CODE REVIEW

**Comment by CrispyBacon1999** [comment]
Looks good, hard to tell what the formatting is like without downloading and building a local copy, but from a look at the rst, I like.

**Comment by prajy** [comment]
Yeah, looks good. Maybe add an example of a subsystem and a command to go with it for command based. Like you wrote shootBall() and intakeBall() so create those 2 commands/subsystems.

File: GoodPractice/code_structure.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
+Code Structure
+==============
+
+Structure of Your Robot Program
+-------------------------------
+It would be wrong of me to say there is one right way of setting up your code. There is not. You can set up your code any way you would like, and as long as you can understand the structure and effectively navigate your code. I am going to explain one structure of code that is useful for teams using SampleRobot, non-command IterativeRobot, PeriodicRobot, and MagicRobot. It is the same structure used by 5 time Innovation in Control Award recipient Team 1418 Vae Victis. ::
```
Comment (endreman0) [inline GoodPractice/code_structure.rst]: Small typo: in the second sentence, remove the first "and".

File: GoodPractice/code_structure.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+robotInit
+~~~~~~~~~
+
+.. note:: The following examples show IterativeRobot, however the same logic an be applied to Sample and Periodic.
+.. tabs::
+
+    .. code-tab:: java
+
+        public class MyRobot(wpilib.Iterative) {
```
Comment (endreman0) [inline GoodPractice/code_structure.rst]: The base class for iterative is `edu.wpi.first.wpilibj.IterativeRobot`.

File: GoodPractice/code_structure.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                drive = Drive(Talon(1), Talon(2));
+            }
+
+        }
+
+    .. code-tab:: py
+
+        class MyRobot(wpilib.IterativeRobot):
+
+            def robotInit():
```
Comment (endreman0) [inline GoodPractice/code_structure.rst]: `self` is not defined - it should be an argument here, right?

File: GoodPractice/code_structure.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+Inside ``robotInit`` is where we actually initializing the variables. There is no real significance to doing this inside ``robotInit`` or when you declare the variables except structure, which is what we're going for. Also notice how I never created variables for the drive motors. If you aren't going to use the variables outside of the drive class, there is no need to declare them as variables here. It makes more sense to declare them as variables inside the drive class, where you can customize them (such as setting PID if they are CANTalons, or reversing them if need be).
+
+.. note:: If you are using RobotMap, this is where the values stored in RobotMap would be used. Instead of ``joystick1 = new Joystick(0)`` you might do ``joystick1 = new Joystick(RobotMap.LEFT_JOYSTICK)``
+
+Teleop
+~~~~~~
+
+Now that you've created all of the robot components, we can focus on telop code. The main basis of teleop code is using ``if`` statements to check for input, and the performing some action based on these events. For example ::
```
Comment (endreman0) [inline GoodPractice/code_structure.rst]: Typo: "telop" ->"teleop"

**Pull request merged**